File: Easy/Java/202. Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        while(n != 1 && n != 4){
            int sum = n;
            n = 0;
            while(sum != 0){
                int temp = sum % 10;
                n += temp * temp;
                sum /= 10;
            }
        }

        return n == 1;
    }
}

File: Easy/Python/202. Happy Number.py
class Solution(object):
    def isHappy(self, n):
        while n != 1 and n != 4:
            this_sum = n
            n = 0
            while this_sum != 0:
                temp = this_sum % 10
                n += temp * temp
                this_sum = this_sum / 10
        
        return n == 1
